### DISCUSSION TITLE
Mods in song select affect star rating when watching replays

### DISCUSSION THREAD

**Discussion by Flamiii** [ppy/osu#18975] [Q&A] [answered]
Map used: https://osu.ppy.sh/beatmapsets/1456709#osu/2993974

When entering gameplay or watching a replay, there's a screen that appears which shows which mods are selected and what the star rating of the map with those mods is. The star rating is correct when you're playing yourself, but if you try to watch a replay, the star rating of the map will show up with the mods that you currently have selected in song select, rather than the ones in the actual replay you're watching.

For example, this is what trying to watch a HDDT replay on Kano - Stella-rium (Asterisk MAKINA Remix) [Starlight] looks like when I have no mods selected in song select:
![osu_2022-07-01_18-37-48](https://user-images.githubusercontent.com/74088315/176977866-9e2ea029-4eaf-45e4-a738-c9458ed8b4d1.png)

If I have EZHT currently selected in song select and then try to watch the HDDT replay:
![osu_2022-07-01_18-41-49](https://user-images.githubusercontent.com/74088315/176977971-74b4aa50-c766-415f-8824-3fc986cbc84a.png)

What it should look like regardless of the mods I have selected in song select (the map with HDDT is actually 8.99*):
![osu_2022-07-01_18-43-00](https://user-images.githubusercontent.com/74088315/176978000-438a9499-af88-4dc1-9a0c-ef27e3cb34ca.png)

**Comment by peppy** [comment, marked as answer]
Tracking via #18980.
